geode-protobuf-client/src/main/java/org/apache/geode/internal/cache/client/protobuf/Region.java

```diff
@@ -0,0 +1,204 @@
+      builder.setLongResult((Long) unencodedValue);
+    } else if (Short.class.equals(unencodedValue.getClass())) {
+      builder.setShortResult((Short) unencodedValue);
+    } else if (Byte.class.equals(unencodedValue.getClass())) {
+      builder.setByteResult((Byte) unencodedValue);
+    } else if (Double.class.equals(unencodedValue.getClass())) {
+      builder.setDoubleResult((Double) unencodedValue);
+    } else if (Float.class.equals(unencodedValue.getClass())) {
+      builder.setFloatResult((Float) unencodedValue);
+    } else if (byte[].class.equals(unencodedValue.getClass())) {
+      builder.setBinaryResult(ByteString.copyFrom((byte[]) unencodedValue));
+    } else if (Boolean.class.equals(unencodedValue.getClass())) {
+      builder.setBooleanResult((Boolean) unencodedValue);
+    } else if (String.class.equals(unencodedValue.getClass())) {
+      builder.setStringResult((String) unencodedValue);
+    }
+    return builder.build();
+  }
+
+  Object decodeValue(BasicTypes.EncodedValue encodedValue) {
+    switch (encodedValue.getValueCase()) {
+      case BINARYRESULT:
+        return encodedValue.getBinaryResult().toByteArray();
+      case BOOLEANRESULT:
+        return encodedValue.getBooleanResult();
+      case BYTERESULT:
+        return (byte) encodedValue.getByteResult();
+      case DOUBLERESULT:
+        return encodedValue.getDoubleResult();
+      case FLOATRESULT:
+        return encodedValue.getFloatResult();
+      case INTRESULT:
+        return encodedValue.getIntResult();
+      case LONGRESULT:
+        return encodedValue.getLongResult();
+      case SHORTRESULT:
+        return (short) encodedValue.getShortResult();
+      case STRINGRESULT:
+        return encodedValue.getStringResult();
+      default:
+        return null;
+    }
+  }
+
+  BasicTypes.Entry encodeEntry(Object unencodedKey, Object unencodedValue) {
+    if (unencodedValue == null) {
+      return BasicTypes.Entry.newBuilder().setKey(encodeValue(unencodedKey)).build();
+    }
+    return BasicTypes.Entry.newBuilder().setKey(encodeValue(unencodedKey))
+        .setValue(encodeValue(unencodedValue)).build();
+  }
+
+  Region(String name, Socket socket) {
+    this.name = name;
+    this.socket = socket;
+  }
+
+  public V get(K key) throws Exception {
+    final OutputStream outputStream = socket.getOutputStream();
+    ClientProtocol.Message.newBuilder()
+        .setRequest(ClientProtocol.Request.newBuilder().setGetRequest(RegionAPI.GetRequest
+            .newBuilder().setRegionName(name).setKey(encodeValue(key.toString()))))
+        .build().writeDelimitedTo(outputStream);
+
+    // TODO: How does one get a java.lang.Object out of
+    // org.apache.geode.internal.protocol.protobuf.v1.EncodedValue?
+    final InputStream inputStream = socket.getInputStream();
+    return (V) ClientProtocol.Message.parseDelimitedFrom(inputStream).getResponse().getGetResponse()
+        .getResult().getStringResult();
```

Comment:
Shouldn't this leverage the decodeValue you wrote earlier:
  return (V) decodeValue(ClientProtocol.Message.parseDelimitedFrom(inputStream).getResponse().getGetResponse().getResult());